src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,31 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.supplyAsync(() -> cache.get(userId), executor);
+        }
+        try {
+            Thread.sleep(1000);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        var data = new UserData(userId, "Details for " + userId);
+        return CompletableFuture.supplyAsync(() -> {
+            cache.put(userId, data);
+            return data;
+        }, executor);
```

Comment:
The current implementation of 'processRequest' method is blocking due to the use of 'Thread.sleep', which is not suitable for an async context. Consider refactoring this method to use a non-blocking approach. For example, you can simulate a delay asynchronously by using 'CompletableFuture.delayedExecutor' along with 'CompletableFuture.supplyAsync'.